core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java

```diff
@@ -565,7 +565,7 @@ private void connectBlock(final Block block, StoredBlock storedPrev, boolean exp
                 StoredBlock commonParent = findSplit(newBlock, chainHead, blockStore);
                 if (chainLockedBlock != null) {
                     StoredBlock cursor = chainHead;
-                    while (cursor != null && !commonParent.getHeader().getHash().equals(cursor) &&
+                    while (cursor != null && !commonParent.getHeader().getHash().equals(cursor.getHeader().getHash()) &&
```

Comment:
this fixes a bug that is related to using chainlocks to be using the correct chain.

The app has to be running when a fork in the blockchain occurs, otherwise the app would not see the fork.